src/main/java/com/notably/logic/commands/NewCommand.java

```diff
@@ -0,0 +1,49 @@
+package com.notably.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import com.notably.commons.core.path.RelativePath;
+import com.notably.commons.core.path.exceptions.InvalidPathException;
+import com.notably.logic.commands.exceptions.CommandException;
+import com.notably.model.BlockManager;
+import com.notably.model.block.Block;
+
+/**
+ * Represent a command used to add a new block.
+ */
+public class NewCommand extends Command {
+    public static final String COMMAND_WORD = "new";
+    private final Block toAdd;
+    private boolean jump;
+
+    public NewCommand(Block block, Boolean jump) {
+        requireNonNull(block);
+        this.toAdd = block;
+        this.jump = jump;
+    }
+
+    /**
+     * Add a new block to the tree structure.
+     * @param notablyModel used to access the tree structure.
+     * @throws CommandException when block of the same Title is detected.
+     */
+    public void execute(BlockManager notablyModel) throws CommandException {
+        if (notablyModel.hasBlock(toAdd)) {
+            throw new CommandException("Block with the same Title detected.");
+        }
+
+        notablyModel.addBlock(toAdd);
+
+        if (this.jump) {
+            try {
+                notablyModel.openBlock(RelativePath.fromString(toAdd.getTitle().toString()));
```

Comment:
If I recall correctly, you wanted to make `NewCommand` only execute the creation, while the open part executed by `OpenCommand` right?

I assume `NewCommandParser` should return a list of `[NewCommand, OpenCommand]` instead of having a `jump` flag here?

Reply:
Hmmm that would clash with the interface that it is currently implementing if i return a List of commands. Or should i create a NewJumpCommand ?

Reply:
I feel creating a `NewJumpCommand` class would be clearer? But not sure if @kevinputera has another way of implementing `NewCommandParser` in mind.

Reply:
Actually, creating a `NewJumpCommand` would result in a very rigid class structure.

Imagine what it would be like if we would enable a user to delete the currently opened note. The intended response is "delete current note" -> "open parent note" right? If that's the case, we would need to also create a `DeleteAndJumpToParent` command, if we were to stick to creating a `NewJumpCommand` now.

How about, instead, you make the `NotablyParser#parseCommand` method return `List<? extends Command>`? The same thing goes for the `CommandParser#parse` method. That way, you will always return a list of commands after every parsing. If there's only one command to execute, you can always return a list containing only one command.

Reply:
Yes i agree and have decided to change the Parser's implementation to return a of Command Instead. Please take note @HemanshuGandhi, @firzanarmani, @kevinputera, @johannagwan.

Reply:
Alrighties, thanks for the clarification! @kevinputera